prov/sm2/src/sm2.h

```diff
@@ -289,6 +289,9 @@ static inline size_t sm2_pop_xfer_entry(struct sm2_ep *ep,
 		return -FI_EAGAIN;
 
 	*xfer_entry = smr_freestack_pop(sm2_freestack(ep->self_region));
+
+	/* ensure we haven't over-exhausted the freestack: */
+	assert(sm2_freestack(ep->self_region)->free >= 0);
```

Comment:
I think this assert would make more sense before you pop
`assert(sm2_freestack(ep->region)->free);`

Reply:
I removed some other checks this commit originally had.  I think now this commit is not so useful.  I'll remove it.